integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java

```diff
@@ -0,0 +1,281 @@
+
+    String[] nodes = {mMasterAddress};
+
+    // Make container request for Tachyon master to ResourceManager
+    ContainerRequest masterContainerAsk =
+        new ContainerRequest(masterResource, nodes, null /* any racks */, priority);
+    LOG.info("Making resource request for Tachyon master on node " + mMasterAddress);
+    mRMClient.addContainerRequest(masterContainerAsk);
+
+    // Wait until Tachyon master container has been allocated
+    while (!mMasterContainerAllocated) {
+      Thread.sleep(1000);
+    }
+
+    // Resource requirements for master containers
+    Resource workerResource = Records.newRecord(Resource.class);
+    workerResource.setMemory(mWorkerMemInMB);
+    workerResource.setVirtualCores(mWorkerCpu);
+
+    // Make container requests for workers to ResourceManager
+    for (int i = 0; i < mNumWorkers; i ++) {
+      ContainerRequest containerAsk = new ContainerRequest(workerResource, null /* any hosts */,
+          null /* any racks */, priority);
+      LOG.info("Making resource request for Tachyon worker " + i + " on any nodes");
+      mRMClient.addContainerRequest(containerAsk);
+    }
+
+    // Wait until all Tachyon worker containers have been allocated
+    while (mNumAllocatedWorkerContainers < mNumWorkers) {
+      Thread.sleep(1000);
+    }
+
+    LOG.info("Master and workers are launched");
+    // Wait for 5 more seconds to avoid application unregistered before some container fully
+    // launched.
+    while (!mApplicationDone) {
+      Thread.sleep(5000);
+    }
+  }
+
+  public void stop() {
+    try {
+      mRMClient.unregisterApplicationMaster(FinalApplicationStatus.SUCCEEDED, "", "");
+    } catch (YarnException yex) {
+      LOG.error("Failed to unregister application " + yex);
+    } catch (IOException ioe) {
+      LOG.error("Failed to unregister application " + ioe);
+    }
+    mRMClient.stop();
+  }
+
+  private void launchTachyonMasterContainers(List<Container> containers) {
+    final String command =
+        new CommandBuilder(PathUtils.concatPath(mTachyonHome, "integration", "bin",
+            "tachyon-master-yarn.sh"))
+            .addArg("1>" + ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/stdout")
+            .addArg("2>" + ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/stderr").toString();
+
+    List<String> commands = Lists.newArrayList(command);
+
+    for (Container container : containers) {
+      try {
+        // Launch container by create ContainerLaunchContext
+        ContainerLaunchContext ctx = Records.newRecord(ContainerLaunchContext.class);
+        ctx.setCommands(commands);
+        LOG.info("Launching container {} for Tachyon master on {} with master command: {}",
+            container.getId(), container.getNodeHttpAddress(), commands);
+        mNMClient.startContainer(container, ctx);
+        String containerUri = container.getNodeHttpAddress(); // in the form of 1.2.3.4:8042
+        mMasterContainerNetAddress = containerUri.split(":")[0];
```

Comment:
Since you can get the master address here, does it matter which node it gets launched on? (do we need `mMasterAddress` passed in still)?